.github/workflows/gh-ci.yaml

```diff
@@ -0,0 +1,81 @@
+name: CI
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**.md'
+  pull_request:
+    paths-ignore:
+      - '**.md'
+
+defaults:
```

Comment:
Put in a concurrency section?

Reply:
Ehh sure, I guess I was mostly fine with default behaviour of tests all dying at once during this introductory phase. There's only one workflow

Reply:
Added

Reply:
with how slow macOS and Windows are, you're probably saving a few trees on multiple commits 🙃